engine/src/resource/random_utils.cpp

```diff
@@ -42,6 +42,6 @@ double randomDouble() {
     std::random_device dev;
     std::mt19937 rng(dev());
     std::uniform_int_distribution<std::mt19937::result_type> int_dist(0,precision);
-    int random_int = int_dist(rng);
-    return (double)random_int/precision;
-}
\ No newline at end of file
+    const unsigned int random_int = int_dist(rng);
```

Comment:
I'm pretty sure that with `precision = 10000`, you can't get negative `random_int`.